### PR TITLE
enable hacbs by default

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/HAC-2492

## Description

- set HACBS FLAG as `true` by default
- HACBS can be disabled by adding to url `hacbs=false`
- HACBS can be disabled by setting localstorage `hacbs` to `true`

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2022-12-06 at 5 11 02 PM

## How to test or reproduce?

Changes HACBS feature flag test

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/hacbs/hacbsFeatureFlag.ts [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
 
 export const HACBS_FLAG = 'HACBS';
+export const IS_HACBS_DEFAULT = 'HACBS';
 
 export const enableHACBSFlagFromQueryParam = (setFlag: SetFeatureFlag): void => {
+  if (IS_HACBS_DEFAULT) {
+    setFlag(HACBS_FLAG, true);
+    return;
+  }
   let enabled = false;
```
Comment (christianvogt) [inline src/hacbs/hacbsFeatureFlag.ts:11]: We just need to set the default value but still allow for overriding via url and local storage.

```suggestion

export const enableHACBSFlagFromQueryParam = (setFlag: SetFeatureFlag): void => {
  let enabled = true;
```

Further down we must also change how we read from localStorage:
```
    enabled = localStorage.getItem('hacbs') === 'true';
```
to:
```
    enabled = localStorage.getItem('hacbs') !== 'false';
```

Comment (abhinandan13jan) [reply src/hacbs/hacbsFeatureFlag.ts]: updated

**Comment by christianvogt** [comment]
@abhinandan13jan do they pass locally for you?

File: integration-tests/tests/hacbs/add-component-to-application-via-hacbs-tab.spec.ts [hunk 2]
```diff
@@ -1,136 +1,166 @@
...
+    pipelineName,
+    'test.appstudio.openshift.io/optional=true',
+    applicationName,
+  ];
+
+  before(function () {
+    localStorage.setItem(LOCAL_STORAGE_KEY_GS_MODAL, 'true');
+    localStorage.setItem(LOCAL_STORAGE_KEY_APPLICATION_MODAL, 'true');
+    // Need to reload the page after enabling HACBS via localStorage
+    cy.reload();
```
Comment (christianvogt) [inline integration-tests/tests/hacbs/add-component-to-application-via-hacbs-tab.spec.ts:40]: Likely don't need this anymore but do need to reload the non-hacbs tests.

**Review by christianvogt — COMMENTED** [review]
Need to update `create-application-from-sample.spec.ts` to disable the `hacbs` flag.

File: integration-tests/tests/hacbs/add-component-to-application-via-hacbs-tab.spec.ts (hunk 2 above)
Comment (abhinandan13jan) [reply integration-tests/tests/hacbs/add-component-to-application-via-hacbs-tab.spec.ts]: Updated

**Comment by abhinandan13jan** [comment]
the HACBS integration test runs fine for me. the hac-dev ones always times-out

Screenshot 2022-12-12 at 5 44 51 PM

**Pull request merged**